packages/schematics/angular/migrations/update-8/differential-loading.ts

```diff
@@ -0,0 +1,98 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import {
+  JsonParseMode,
+  isJsonObject,
+  join,
+  normalize,
+  parseJson,
+  parseJsonAst,
+} from '@angular-devkit/core';
+import { Rule, Tree } from '@angular-devkit/schematics';
+import {
+  findPropertyInAstObject,
+  insertPropertyInAstObjectInOrder,
+} from '../../utility/json-utils';
+
+export function updateES5Projects(): Rule {
+  return (host: Tree) => {
+    const tsConfigPath = '/tsconfig.json';
+    const buffer = host.read(tsConfigPath);
+    if (!buffer) {
+      return host;
+    }
+
+    const tsCfgAst = parseJsonAst(buffer.toString(), JsonParseMode.Loose);
+
+    if (tsCfgAst.kind !== 'object') {
+      return host;
+    }
+
+    const compilerOptions = findPropertyInAstObject(tsCfgAst, 'compilerOptions');
+    if (!compilerOptions || compilerOptions.kind !== 'object') {
+      return host;
+    }
+
+    const scriptTarget = findPropertyInAstObject(compilerOptions, 'target');
+    if (scriptTarget && scriptTarget.value === 'es2015') {
+      return host;
+    }
+
+    const recorder = host.beginUpdate(tsConfigPath);
+    if (scriptTarget) {
+      const { start, end } = scriptTarget;
+      recorder.remove(start.offset, end.offset - start.offset);
+      recorder.insertLeft(start.offset, '"es2015"');
+    } else {
+      insertPropertyInAstObjectInOrder(recorder, compilerOptions, 'target', 'es2015', 4);
+    }
+    host.commitUpdate(recorder);
+
+    return updateBrowserlist;
+  };
+}
+
+function updateBrowserlist(): Rule {
+  return (tree) => {
+    const angularConfigContent = tree.read('angular.json') || tree.read('.angular.json');
+
+    if (!angularConfigContent) {
+      return;
+    }
+
+    const angularJson = parseJson(angularConfigContent.toString(), JsonParseMode.Loose);
+
+    if (!isJsonObject(angularJson) || !isJsonObject(angularJson.projects)) {
+      // If that field isn't there, no use...
+      return;
+    }
+
+    // For all projects
+    for (const projectName of Object.keys(angularJson.projects)) {
+      const project = angularJson.projects[projectName];
+      if (!isJsonObject(project)) {
+        continue;
+      }
+      if (typeof project.root != 'string' || project.projectType !== 'application') {
+        continue;
+      }
+
+      const browserslistPath = join(normalize(project.root), '/browserslist');
+      const source = tree.read(browserslistPath);
+      if (!source) {
```

Comment:
What does it mean to not have the source? Should we create the browserlist in this case?

Reply:
It means, that it will fallback to the default browserlist configuration or it can mean that a users specified the browserlist configuration in another way. 

I think creating this file might be a good approach

Reply:
Sounds good. If we create the file, are we going to overwrite the user's configuration if they are not using the defaults? This might be a bit frustrating for them, I guess?

Reply:
Only if they are using `.browserslistrc`

Reply:
Only if they are using `.browserslistrc`